src/bsp/src/stm32/include/HostUart.h

```diff
@@ -0,0 +1,49 @@
+#ifndef __HOSTUART_H__
+#define __HOSTUART_H__
+
+#include "bsp/ICRC.h"
+#include "bsp/IHostUart.h"
+#include <FreeRTOS.h>
+#include <cstdint>
+#include <logger/ILogger.h>
+#include <semphr.h>
+
+class HostUart : public IHostUart {
+  public:
+    explicit HostUart(ICRC& crc, ILogger& logger);
+    ~HostUart() override = default;
+
+    bool send(const uint8_t* buffer, uint16_t length) override;
+    bool isBusy() override;
+
+    void process();
+
+    friend void hostUart_C_txCpltCallback(void* hostUartInstance);
+    friend void hostUart_C_rxCpltCallback(void* hostUartInstance);
+
+  private:
+    enum RxState { Rx_Idle, Rx_WaitForHeader, Rx_WaitForPayload, Rx_CheckIntegrity };
+    enum TxState { Tx_Idle, Tx_SendHeader, Tx_SendPayload };
```

Comment:
use `enum class` instead, they don't overlap eachother, or if a global enum has the same name

Reply:
Though I saw you are using :: notation for access, maybe it's already an enum class since it's in defined in a class

Reply:
Changed for enum class